src/Compilers/CSharp/Portable/BoundTree/DecisionTreeBuilder.cs

```diff
@@ -116,7 +116,12 @@ protected DecisionTree AddToDecisionTree(DecisionTree decisionTree, SyntaxNode s
                     {
                         var declarationPattern = (BoundDeclarationPattern)pattern;
                         DecisionMaker maker =
-                            (e, t) => new DecisionTree.Guarded(e, t, ImmutableArray.Create(new KeyValuePair<BoundExpression, BoundExpression>(e, declarationPattern.VariableAccess)), sectionSyntax, guard, label);
+                            (e, t) => new DecisionTree.Guarded(
+                                        e, t,
+                                        (declarationPattern.VariableAccess == null)
+                                            ? ImmutableArray<KeyValuePair<BoundExpression, BoundExpression>>.Empty
```

Comment:
Looks like this API is fully internal: let's replace KVP with a tuple type.

Reply:
The decision tree and all its machinery will be going away, to be replaced by the DecisionDag and a different mechanism for constructing it which you can see in DecisionDagBuilder. As such, I do not intend to do any refactoring to this class except to keep it working until it is removed.

---
In reply to: [155095151](https://github.com/dotnet/roslyn/pull/23209#discussion_r155095151) [](ancestors = 155095151)